src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useContext } from 'react';
+import { TodoContext } from '../../context/TodoContext';
+import { deleteTodos } from '../../api/todos';
+import { Error } from '../../types/Error';
+import { TodoFilter } from '../TodoFilter';
+
+type Props = {
+  activeTodosAmount: number;
+  completedTodosIds: number[];
+};
+
+export const Footer: React.FC<Props> = ({
+  activeTodosAmount,
+  completedTodosIds,
+}) => {
+  const {
+    deleteTodo,
+    handleUpdatingTodosIds,
+    handleError,
+  } = useContext(TodoContext);
+
+  const handleDeleteTodos = () => {
+    handleError('');
+
+    completedTodosIds.forEach(id => {
+      handleUpdatingTodosIds(id);
+      deleteTodos(id)
+        .then(() => deleteTodo(id))
+        .catch(() => {
+          handleError(Error.Delete);
+        })
+        .finally(() => handleUpdatingTodosIds(null));
```

Comment:
check your app, you use this code several time

```
  deleteTodos(id)
        .then(() => deleteTodo(id))
        .catch(() => {
          handleError(Error.Delete);
        })
        .finally(() => handleUpdatingTodosIds(null));
```

try to move it to context delete function and reuse in all places you need

Reply:
I placed wrong demo link for this tusk, sorry, fixed already)